packages/flutter/lib/src/cupertino/text_form_field_row.dart

```diff
@@ -327,13 +329,11 @@ class _CupertinoTextFormFieldRowState extends FormFieldState<String> {
 
   @override
   void reset() {
+    // Set the controller value before calling super.reset() to let
+    // _handleControllerChanged suppress the change.
+    _effectiveController!.text = widget.initialValue!;
     super.reset();
-
-    if (widget.initialValue != null) {
-      setState(() {
-        _effectiveController!.text = widget.initialValue!;
-      });
-    }
+    _cupertinoTextFormFieldRow.onChanged?.call(_effectiveController!.text);
```

Comment:
Should we still call this if the value did not change?

Reply:
Same for other `FormField`s.

Reply:
I chose not to add such a check in order to be consistent with `FormState.reset` which (indirectly) call the `Form.onChanged` callback without checking if field values have changed.
I also checked that `DropdownButton.onChanged` is called when the same entry is selected.

Maybe we should consider this is another issue/PR? I don't know if this was done on purpose.